### ISSUE TITLE
submit_batch_jobs.sh significant figures and repeat SNe not getting merged

### ISSUE THREAD

**Issue by erikpeterson23** [RickKessler/SNANA#368]
Working with Dan Scolnic on the submit_batch_jobs.sh tool.

I've placed an example here:
`$SNANA_DEBUG/epeterson3`

Calling
`submit_batch_jobs.sh plus_erik.input --ignore_fitopt`

I find 2 issues in the outputs: (which go here: `/scratch/midway2/rkessler/PIPPIN_OUTPUT/PLUS_erik/OUTPUT_BBCFIT/`)
1.) although the header overrides have heliocentric redshifts up to 8 decimal places, my output files have 3 decimal places.
ex. (example files in `$SNANA_DEBUG/epeterson3`)
`FITOPT000_MUOPT003.FITRES.gz` has 2007bd zHEL=0.030
while my header override `pos_lowz_50.txt` has 2007bd zHEL=0.0304675
All heliocentric redshifts seem to do this.

2.) although I have `iflag_duplicate = 2`, my output files have variable numbers of repeated SNe
ex.
`FITOPT000_MUOPT003.FITRES.gz` has 2009ig repeated 4 times.
`FITOPT000_MUOPT004.FITRES.gz` has 2009ig repeated 1 time.
`FITOPT000_MUOPT003.FITRES.gz` has 2008Q repeated 1 time.
`FITOPT000_MUOPT004.FITRES.gz` has 2008Q repeated 2 times.

Erik Peterson

**Comment by RickKessler** [comment]
Fixed bug for item 1); all z-variables from override table will be written with %.5f format.

**Comment by RickKessler** [comment]
For item 2) there was a limit of 20 sets of duplicates and your job has ~ 100. The abort trap had a bug so didn't abort on exceeding the limit. Previous analyses should take notice of this bug. Increased limit to 200, and fixed abort trap.

**Comment by erikpeterson23** [comment]
For item 2), duplicate SNe still persist (albeit fewer SNe make it through).

I copied updated outputs into `$SNANA_DEBUG/epeterson3` after rerunning.
In the outputs I copied,
`FITOPT_MUOPT003.FITRES.gz` has `2008Q` repeated 1 time.
`FITOPT_MUOPT004.FITRES.gz` has `2008Q` repeated 2 times.
`FITOPT_MUOPT003.FITRES.gz` has `2013ex` repeated 2 times.
`FITOPT_MUOPT004.FITRES.gz` has `2013ex` repeated 1 time.

`2011by` is the only other SN that remains duplicated in any of my other outputs.
(which get outputted here: `/scratch/midway2/rkessler/PIPPIN_OUTPUT/PLUS_erik/OUTPUT_BBCFIT/`)

**Comment by RickKessler** [comment]
2008Q and 2000cx have exactly the same redshift and same VPEC, so I suspect a bookkeeping issue here ... unless they have the same host ? The SALT2mu code sorts by redshift to find duplicates, and 2000cx likely got in the middle of the two 2008Q entries. 
<img width="627" alt="lowz-duplicates" src="https://user-images.githubusercontent.com/24592476/106166202-89f96400-6151-11eb-9569-b1dba344f8ea.png">

**Comment by RickKessler** [comment]
I just checked PKMJD, and they differ by almost 8 years as indicated by their IAUC names. Please check that they are from the same host, and I will add PKMJD window to the matching requirement.

**Comment by RickKessler** [comment]
The RA and DEC differ by 0.04 degrees (140'') ... so not sure it's the same galaxy??

**Comment by RickKessler** [comment]
I have re-worked the duplicate logic to allow for different events in the same galaxy. Try again and please run your independent duplicate-checker.

**Comment by erikpeterson23** [comment]
I was able to rerun this morning. On this attempt, it seems that there are more SNe repeated than before. I have copied updated outputs to `$SNANA_DEBUG/epeterson3`.

Examples include:
`2009Y` repeated twice in`FITOPT000_MUOPT003.FITRES.gz` while only once in `FITOPT000_MUOPT004.FITRES.gz`
`2008gl` repeated twice in`FITOPT000_MUOPT003.FITRES.gz` while only once in `FITOPT000_MUOPT004.FITRES.gz`
`2009fv` repeated twice in`FITOPT000_MUOPT003.FITRES.gz` while only once in `FITOPT000_MUOPT004.FITRES.gz`

Other MUOPTs have different SNe repeated variably (not just 003 and 004).

For my full analysis I run `submit_batch_jobs.sh plus_erik.input --ignore_fitopt`
here: `/project2/rkessler/SURVEYS/DES/USERS/epeterson3/pec_vel`
and the outputs go here: `/scratch/midway2/rkessler/PIPPIN_OUTPUT/PLUS_erik/OUTPUT_BBCFIT`

I ran on `plus_erik.input` as well within `$SNANA_DEBUG/epeterson3` and confirmed the same example issues in `OUT_RICK/OUTPUT_BBCFIT`.
